### DISCUSSION TITLE
"RecursionError: maximum recursion depth exceeded" when extending header

### DISCUSSION THREAD

**Discussion by mateus2k2** [squidfunk/mkdocs-material#7267] [Q&A] [answered]
Hi, When I try to do:

```
{% extends "base.html" %}

{% block announce %}
  <!-- Add announcement here, including arbitrary HTML -->
{% endblock %} 
```

I get the error:

```
  File "<frozen _collections_abc>", line 620, in __rsub__
  File "<frozen _collections_abc>", line 830, in _from_iterable
  File "<frozen _collections_abc>", line 621, in <genexpr>
  File "<frozen _collections_abc>", line 833, in __contains__
  File "/usr/local/lib/python3.11/collections/__init__.py", line 1021, in __contains__
    return any(key in m for m in self.maps)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RecursionError: maximum recursion depth exceeded
ERROR   -  [21:22:44] An error happened during the rebuild. The server will appear stuck until build errors are resolved.
```

**Comment by kamilkrzyskow** [comment]
Hello @mateus2k2,
change the name of the overriden template to `main.html`, as you're extending `base.html`, so naming it the same creates an infinite loop.
Please re-read the section in the docs: https://squidfunk.github.io/mkdocs-material/customization/#overriding-blocks

Also please use the search feature before starting new discussions, duplicate of:
- https://github.com/squidfunk/mkdocs-material/discussions/5169

**Reply by mateus2k2** [reply]
I'am sorry, but I am getting confused. This is my structure:

![image](https://github.com/squidfunk/mkdocs-material/assets/24702324/11486e14-6602-4f9c-8b6f-0b2b90c6c650)

main.html
```
{% extends "base.html" %}

{% block htmltitle %}
  <title>Lorem ipsum dolor sit amet</title>
{% endblock %} 
```

header.html
```
{% extends "base.html" %}
```

changing it to {% extends "base.html" %} or {% extends "main.html" %} does not work on header.html
Only {% extends "base.html" %} works on the main.html

**Reply by kamilkrzyskow** [reply, marked as answer]
Oh, then perhaps this isn't a duplicate, sorry.

You can only override blocks in `main.html`. If you're overriding partials, you can't extend `base.html` or `main.html`.
It is executed from top to bottom `main.html` -> `base.html` -> `partials/*.html` it can't go back up.

This is the allowed structure in `header.html`:
with comments: https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/partials/header.html
minified: https://github.com/squidfunk/mkdocs-material/blob/master/material/templates/partials/header.html

If you want to override the announcement block then do it in main.html

**Reply by mateus2k2** [reply]
I see, to thought I used {% extends "base.html" %} in the header.html to keep the header as is and add stuff on top. But I think I understand now, thanks
